### PR TITLE
feat: Types and Intellisense Improvements

### LINKED ISSUE #188: feat(common): types and autocompletion
We want to do some exploratory investigation to see how we can improve developer experience with regards to using our packages in terms of the types we provide and autocompletion on VS Code

### PR DESCRIPTION
**Pull request by fazzaamiarso**
## feat: Types and Intellisense Improvements

Fixes #188 

Description of changes: 
- Add JSDocs to packages' exported API/methods 

Tag a reviewer: @ayoayco @Icyscools 

Tasks:
- [x] I have ran the build command to make sure apps are working: `npm run build`
- [x] I have ran the tests to make sure nothing is broken: `npm run test`
- [x] I have ran the linter to make sure code is clean: `npm run lint`
- [x] I have reviewed my own code to remove changes that are not needed

### CODE REVIEW

**Comment by ayoayco** [comment]
Hey @fazzaamiarso thanks for this! I found a good API documentation (ng form) we can take inpiration from :) https://angular.io/api/forms -- can you see their wording if it fits our APIs too?

File: packages/form/core/form-control.ts [hunk 1]
```diff
@@ -135,18 +141,29 @@ export class FormControl {
...
+	 * @param validators - array of `Validators` return value
+	 */
 	setValidators(validators: string[]) {
 		this._validators = validators;
 		const valueStr: string = this._value?.toString() || '';
 		this._errors = this.validate(valueStr, this._validators || []);
 	}
 
+	/**
+	 * Set if a control should be validated and rendered on server side
```
Comment (ayoayco) [inline packages/form/core/form-control.ts:165]:
```suggestion
	 * Sets the property that determines if control will be validated before rendering on the server
```

**Comment by fazzaamiarso** [comment]
> Hey @fazzaamiarso thanks for this! To me this looks a good start already. Though I found a good API documentation (ng form) we can take inpiration from :) https://angular.io/api/forms#structures -- can you see their wording if it fits our APIs too?
> 
> Also, realized I wrote some description on [our docs](https://docs.astro-reactive.dev/en/api/form/form-component/)! Which I guess I should have copied as TSdoc too 😅

Will check it!

File: packages/form/core/form-control.ts [hunk 2]
```diff
@@ -41,6 +44,9 @@ export class FormControl {
 		return [];
 	};
 
+	/**
+	 * Create a form control object to build form element
```
[the remark this answers is not in the document]
Comment (fazzaamiarso) [reply packages/form/core/form-control.ts:48]: It appeared 👍

**Review by ayoayco — APPROVED** [review]
Great job @fazzaamiarso!!

**Pull request merged**